docs/admin/kubeadm-init.md

```diff
@@ -0,0 +1,425 @@
+a future version. To create a self-hosted cluster, pass the `--feature-gates=SelfHosting=true`
+flag to `kubeadm init`.
+{: .caution}
+
+**Warning:** Read carefully current caveats/limitations. 
+{: .warning}
+
+#### Caveats
+
+Kubeadm self-hosting in 1.8 has some important limitations. In particular, a
+self-hosted cluster cannot currently recover from a reboot of the master node
+without manual intervention. This and other limitations are expected to be
+resolved before self-hosting graduates from alpha.
+
+By default, self-hosted control plane pods rely on credentials loaded from
+[`hostPath`](https://kubernetes.io/docs/concepts/storage/volumes/#hostpath)
+volumes. Except for initial creation, these credentials are not managed by
+kubeadm. You can use `--feature-gates=StoreCertsInSecrets=true` to enable an
+experimental mode where control plane credentials are loaded from Secrets
+instead. This requires very careful control over the authentication and
+authorization configuration for your cluster, and may not be appropriate for
+your environment.
+
+In 1.8, the self-hosted portion of the control plane does not include etcd,
+which still runs as a static pod.
+
+#### Process
+The self-hosting bootstrap process is documented in [the kubeadm 1.8 design
+document](https://github.com/kubernetes/kubeadm/blob/master/docs/design/design_v1.8.md#optional-self-hosting).
+In summary, `kubeadm init --feature-gates=SelfHosting=true` works as follows:
+
+  1. As usual, kubeadm creates static pod YAML files in `/etc/kubernetes/manifests/`.
+
+  1. Kubelet loads these files and launches the bootstrap static control plane.
+    Kubeadm waits for this bootstrap static control plane to be running and
+    healthy. This is identical to the `kubeadm init` process without self-hosting.
+
+  1. Kubeadm uses the static control plane pod manifests to construct a set of
+    DaemonSet manifests that will run the self-hosted control plane.
+    It also modifies these manifests where necessary, for example adding new volumes 
+    for secrets.
+
+  1. Kubeadm creates DaemonSets in the `kube-system` namespace and waits for the
+     resulting pods to be running.
+
+  1. kubeadm then waits for a self-hosted pod to be running. Once it's operational, 
+     it's associated static pod is deleted and kubeadm moves on to install the next 
+     component. This triggers kubelet to stop those static pods.
+
+  1. When the original static control plane stops, the new self-hosted control
+    plane is able to bind to listening ports and become active.
+
+This process (steps 3-6) can also be triggered with `kubeadm phase selfhosting convert-from-staticpods`.
+
+### Running kubeadm without an internet connection
+
+All of the control plane components run in Pods started by the kubelet and
+the following images are required for the cluster works will be automatically
+pulled by the kubelet if they don't exist locally while `kubeadm init` is initializing
+your master:
+
```

Comment:
Take a look at "and the following images ...is initializing your master". It seems like the sentence got out of control.